src/js/popup.js

```diff
@@ -1359,6 +1365,113 @@ Logic.registerPanel(P_CONTAINER_EDIT, {
     }
   },
 
+  async _addSite() {
+    // Get URL and container ID from form
+    const formValues = new FormData(this._editForm);
+    const url = formValues.get("site-name");
+    const userContextId = formValues.get("container-id");
+    const currentTab = await Logic.currentTab();
+    const tabId = currentTab.id;
+    const baseURL = this.normalizeUrl(url);
```

Comment:
I'm pretty sure normalization isn't necessary. Assuming the below call to `Logic.setOrRemoveAssignment` should be `Utils.setOrRemoveAssignment`,
- `Utils.setOrRemoveAssignment` send a message which https://github.com/mozilla/multi-account-containers/blob/50f5ebfcffaa0d2d09b9199de7173f3e6ce25629/src/js/utils.js#L93-L99
- the anonymous function in `messageHandler.init` handles by calling https://github.com/mozilla/multi-account-containers/blob/50f5ebfcffaa0d2d09b9199de7173f3e6ce25629/src/js/background/messageHandler.js#L49
- `assignManager._setOrRemoveAssignment` which calls https://github.com/mozilla/multi-account-containers/blob/50f5ebfcffaa0d2d09b9199de7173f3e6ce25629/src/js/background/assignManager.js#L560-L563
- `storageArea.set` which filters the URL through https://github.com/mozilla/multi-account-containers/blob/50f5ebfcffaa0d2d09b9199de7173f3e6ce25629/src/js/background/assignManager.js#L73
- `storageArea.getSiteStoreKey` which extracts the hostname and possibly the port from the URL https://github.com/mozilla/multi-account-containers/blob/50f5ebfcffaa0d2d09b9199de7173f3e6ce25629/src/js/background/assignManager.js#L14-L20

Reply:
I'm fairly new to this, so any help you can give - perhaps a PR into my patch - would be very much appreciated.

Reply:
Isn't the point of `nomarlizeUrl` to trim the path off? Given that it's random user input, this seems like a fairly sensible idea.

It looks like you're right about the `Utils.setOrRemoveAssignment` since it's only defined in [src/js/utils.js](https://github.com/mozilla/multi-account-containers/blob/main/src/js/utils.js#L128), so I've committed that change too.